core/producer.go

```diff
@@ -59,21 +60,85 @@ func (status SendStatus) String() string {
 	}
 }
 
-func newDefaultProducer(config *ProducerConfig) *defaultProducer {
+func newDefaultProducer(config *ProducerConfig) (*defaultProducer, error) {
+	if config.GroupID == "" {
```

Comment:
Maybe judge the config with nil?

Reply:
In golang, the default value of `string` type is `""` not `nil`

Reply:
fixed